---
title: "RNASeqR: an R package for automated two-group RNA-Seq
analysis workflow"
collection: publications
permalink: /publication/2019_12_04-RNASeqR
excerpt: 'This R package is designed for case-control RNA-Seq analysis (two-group). There are six steps: "RNASeqRParam S4 Object Creation", "Environment Setup", "Quality Assessment", "Reads Alignment & Quantification", "Gene-level Differential Analyses" and "Functional Analyses". Each step corresponds to a function in this package. After running functions in order, a basic RNASeq analysis would be done easily.'
date: 2019-12-04
venue: 'IEEE/ACM Transactions on Computational Biology and Bioinformatics (TCBB)'
paperurl: 'https://ieeexplore.ieee.org/document/8918337'
citation: '<br><b style="color:#ad0000">K.H. Chao*</b>, Y.W. Hsiao, Y.F. Lee, C.Y. Lee, L.C. Lai, M.H. Tsai, T.P. Lu, and E.Y. Chuang (2019). "RNASeqR: an R package for automated two-group RNA-Seq analysis workflow" in <b><i>IEEE/ACM Transactions on Computational Biology and Bioinformatics (TCBB)</i></b>. doi: 10.1109/TCBB.2019.2956708.'
---
---

## Abstract

RNA-Seq analysis has revolutionized researchersâ€™ understanding of the transcriptome in biological research. Assessing the differences in transcriptomic profiles between tissue samples or patient groups enables researchers to explore the underlying biological impact of transcription. RNA-Seq analysis requires multiple processing steps and huge computational capabilities. There are many well-developed R packages for individual steps; however, there are few R/Bioconductor packages that integrate existing software tools into a comprehensive RNA-Seq analysis and provide fundamental end-to-end results in pure R environment so that researchers can quickly and easily get fundamental information in big sequencing data. To address this need, we have developed the open source R/Bioconductor package, RNASeqR. It allows users to run an automated RNA- Seq analysis with only six steps, producing essential tabular and graphical results for further biological interpretation. The features of RNASeqR include: six-step analysis, comprehensive visualization, background execution version, and the integration of both R and command-line software. RNASeqR provides fast, light-weight, and easy-to-run RNA-Seq analysis pipeline in pure R environment. It allows users to efficiently utilize popular software tools, including both R/Bioconductor and command-line tools, without predefining the resources or environments. RNASeqR is freely available for Linux and macOS operating systems from Bioconductor (https://bioconductor.org/packages/release/bioc/html/RNASeqR.html).

## Conclusion
RNASeqR is a new Bioconductor package providing a six- step automated workflow for two-group comparative RNA-Seq analysis. The core design concept of RNASeqR is that each RNA-Seq analysis step is implemented as an R function in the package, and thus users can perform RNA- Seq analysis instinctively. The main features of RNASeqR include: (i) flexible function options in both an R-interac- tive version and a background version, (ii) comprehensive visualization, (iii) integration of command-line tools into the R environment, (iv) a well-defined file structure, and (v) ease of use. We believe that RNASeqR will assist clinical researchers without significant computational background to obtain useful information from RNA-Seq data in an easy, efficient and reproducible manner.


---
**Contact:**&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kuanhao.chao@gmail.com

**Index Terms:** &nbsp;&nbsp;RNA-Seq, Analysis workflow, Pipeline, R, Bioconductor, Transcriptome assembly, Differential expression analysis, Gene expression, Statistical analysis, Visualization

[Download paper here](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8918337&isnumber=4359833)
